tests/integration/test_dynamic_sampling.py

```diff
@@ -583,18 +583,33 @@ def test_multi_item_envelope(mini_sentry, relay, rule_type, event_factory):
 def test_client_sample_rate_adjusted(mini_sentry, relay, rule_type, event_factory):
     """
     Tests that the client sample rate is honored when applying server-side
-    sampling. Do so by sending an envelope with a very low reported client sample rate
-    and a sampling rule with the same sample rate. The server should adjust
-    itself to 1.0. The chances of this test passing without the adjustment in
-    place are very low (but not 0).
+    sampling.
+
+    When client_sample_rate=SAMPLE_RATE, the server should adjust to let most
+    events through (since client already sampled). When client_sample_rate=1.0,
+    the server should apply the full sampling rate, dropping most of the events.
     """
 
-    def assert_client_reports():
-        # Relay is sending a client report, skip over it (1 for transactions, 1 for spans)
-        for _ in range(2):
-            received_envelope = mini_sentry.captured_events.get(timeout=1)
-            assert received_envelope.get_transaction_event() is None
-            assert received_envelope.get_event() is None
+    def collect_events_batch(expected_count, timeout_per_event=0.1):
+        events = []
```

Comment:
This could be a counter, right? We only append to it and return the length at the end.

Reply:
yeah, I was first asserting what events I got if any, but changed that without actually removing the list. Good point , thanks!